vignettes/unit_test_guidance.Rmd

```diff
@@ -0,0 +1,151 @@
+with lower complexity is easier to understand and maintain, which means developers 
+are less likely to introduce bugs when working on it. We can measure the 
+cyclomatic complexity of your tests (using, for instance, a linter tool) and do 
+your best to keep it low.
+
+### Test Shouldn’t Duplicate Implementation Logic
+If the same person wrote both the test and the implementation, it’s possible they
+made the same errors in both places. Since the tests mirror the implementation,
+they might still pass, and the implementation could be wrong, but the tests might
+fool you into thinking otherwise. Resist the urge to make your tests fancy, keep
+them simple, and your testing suite will be better for it.
+
+### Tests Should Be Readable
+This best practice overlaps a little bit with the one about keeping your tests 
+simple. If tests are hard to read, developers are more likely to misunderstand 
+them and introduce bugs. Test cases could be used as a form of documentation, so
+they obviously need to be readable.
+
+### Running Unit Tests Part of the Build Process
+Automate the whole process of running the unit tests and taking some action when
+they fail. Your build process should execute your unit tests and mark the build 
+as broken when the tests fail.
+
+# Writing Unit Tests in Admiral
+
+## Plan your Unit Tests
+Start by considering the derivation rule you are testing and the possible 
+arguments/flexibilities of your function code. Then plan which scenarios you will
+test. These can either involve generating different input test cases or feeding 
+them into different calls of your function.
+
+## Tests Should be Robust to Cover Realistic Data Scenarios
+For generating input test cases, it can be helpful to consider regular cases 
+(expected common data scenarios), boundary cases (where data points are close or
+equal), and special cases (uncommon but valid data scenarios, e.g. missing or 
+special characters). Although you will never cover every single eventuality of 
+possible input data (no reliability testing method ever gives 100% certainty), 
+you do need to give confidence that the code is robust enough to work across most
+data scenarios.
+
+## Testing Should Cover Possible Arguments
+For the different calls of your function, consider how the user might apply your
+function and test a variety of possible calls, whilst still remembering the tips
+above that tests should be fast and simple. 
+This is only needed in cases where the complexity and level of flexibility of 
+your function justifies it, e.g. see the test script: https://github.com/Roche-GSK/admiral/blob/master/tests/testthat/test-derive_extreme_flag.R.
+
+## Exported Functions
+Don’t forget to add a unit test for each exported function.
+
+## Set up the Test Script
+Within {admiral} folder https://github.com/Roche-GSK/admiral/tree/master/tests/testthat, 
+add a script with the naming convention “test-<function name>.R” and use the 
+following format:
```

Comment:
I would slightly reoword. The test file name should have the same name as the implementation file. If there are multiple function in `some_funs.R` file then these should all be tested in `test-some_funs.R`.

Reply:
Done, added the extra piece about creating the unit test file from console.